pio/lib/Sender/Sender.cpp

```diff
@@ -485,3 +487,237 @@ bool SenderClass::sendTCONTROL(String server, uint16_t port)
     stopclient();
     return true;
 }
+
+//Blynk HTTP was taking 2 seconds longer and did not show in the App
+//when device was connected, therefore best to use their API.
+bool SenderClass::sendBlynk(char* token)
+{
+    serializeJson(_doc, Serial);
+
+    Blynk.config(token);
+
+    byte i = 0;
+    int _pin = 0;
+    String _value;
+
+    while (!Blynk.connected() && i<100)
+    {
+      Blynk.run();
+      i++;
+      delay(50);
+    }
+        
+    if (Blynk.connected())
+    {
+        CONSOLELN(F("\nConnected to the Blynk server, sending data"));
+
+        for (const auto &kv : _doc.as<JsonObject>())
+        {
+            _pin = atoi(kv.key().c_str());
+            _value = kv.value().as<String>();
+            Blynk.virtualWrite(_pin, _value);
+        }
+    }
+
+    else {
+        CONSOLELN(F("\nFailed to connect to Blynk, going to sleep"));
+        return false;
+    }
+    
+    delay(300);     //delay to allow last value to be sent;
```

Comment:
That’s obsolete, is been taken care of at the main caller

Reply:
Not sure what is obsolete here, can you please clarify?

Reply:
The delay here is covered already in the main routine, doesn’t need duplication

Reply:
This is required for Blynk for some reason, without it the last value is not sent. I tested with 100ms and it works. So I'm afraid we need to leave it there.